cmd/connection.go

```diff
@@ -0,0 +1,601 @@
+		options.format, options.filter, options.fileName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+func accessRoleShowCmd() *cobra.Command {
+	options := connectionOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "roles",
+		Short: "Get saved access roles for a connection",
+		Long:  `Get saved access roles for a connection`,
+		Example: `
+	privx-cli connections roles [access flags] --id <CONN-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return accessRoleShow(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.connID, "id", "", "connection ID")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func accessRoleShow(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	roles, err := api.AccessRoles(options.connID)
+	if err != nil {
+		return err
+	}
+
+	return stdout(roles)
+}
+
+//
+//
+func connectionAccessRoleGrantCmd() *cobra.Command {
+	options := connectionOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "grant-access-role-permission",
+		Short: "Grant a permission for a role for a connection",
+		Long:  `Grant a permission for a role for a connection`,
+		Example: `
+	privx-cli connections grant-role-permission [access flags] --id <CONN-ID> --rid <ROLE-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return connectionAccessRoleGrant(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.connID, "id", "", "connection ID")
+	flags.StringVar(&options.roleID, "rid", "", "role ID")
+	cmd.MarkFlagRequired("id")
+	cmd.MarkFlagRequired("rid")
+
+	return cmd
+}
+
+func connectionAccessRoleGrant(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	err := api.GrantAccessRoleToConnection(options.connID, options.roleID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+func connectionAccessRoleRevokeCmd() *cobra.Command {
+	options := connectionOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "revoke-role-from-connection",
+		Short: "Revoke a permission for a role from a connection",
+		Long:  `Revoke a permission for a role from a connection`,
+		Example: `
+	privx-cli connections revoke-role-from-connection [access flags] --id <CONN-ID> --rid <ROLE-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return connectionAccessRoleRevoke(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.connID, "id", "", "connection ID")
+	flags.StringVar(&options.roleID, "rid", "", "role ID")
+	cmd.MarkFlagRequired("id")
+	cmd.MarkFlagRequired("rid")
+
+	return cmd
+}
+
+func connectionAccessRoleRevoke(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	err := api.RevokeAccessRoleFromConnection(options.connID, options.roleID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+func allConnectionAccessRoleRevokeCmd() *cobra.Command {
```

Comment:
The way I imagine the usage is:
```
# Revoke the role from all connections
privx-cli connections revoke-access-role --role-id <role_id>
# Revoke access role from one connection
privx-cli connections revoke-access-role --conn-id <connection_id> --role-id <role_id>
```

Obviously we still have 2 different functions and branch off the execution based on the values of the flags. Note that since we are working with 2 resources I preferred to refer to --conn-id  instead of just --id. I recommend you to do the same if that is OK.